File: python/Level 2/멀쩡한_사각형.py
def gcd(a, b):
    while True:
        if a > b:
            large, small = a, b
        else:
            large, small = b, a

        if large % small == 0:
            return small

        a = large-small
        b = small


def solution(w, h):
    d = gcd(w, h)
    return w*h - (w//d + h//d - 1)*d

File: golang/Level 2/멀쩡한_사각형.go
func gcd(a int, b int) int {
    var large int
    var small int
    
    for {
        if a > b {
            large = a
            small = b
        } else {
            large = b
            small = a
        }
        if large % small == 0 {
            return small
        }
        a = large - small
        b = small
    }
}

func solution(w int, h int) int64 {
    var answer int64
    d := gcd(w, h)
    answer = int64(w*h) - (int64(w/d) + int64(h/d) - 1)*int64(d)
    return answer
}